### PR TITLE
[Library Supports Union -- PR 2/TBD] Deserialization functions

### PR DESCRIPTION
**Pull request by rkhaled98**
NOTE: this PR is part 2/TBD of a series of upcoming PRs and as such some tests will be dependent on code in other branches. Breaking up into multiple PRs was done for reviewer ease.

- [x] write deserializer to convert both yaml files and strings into the "spec tree" which is simply a LinkedHashMap<String, Object>
- [x] 100% test coverage

### CODE REVIEW

File: library/src/main/java/YamlStringToSpecTreeConverter.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import java.util.LinkedHashMap;
+import org.yaml.snakeyaml.Yaml;
+
+/** Provides static functions for converting YAML files and strings into Maps */
+public class YamlStringToSpecTreeConverter {
+
+  /**
+   * deserializes a YAML file at the given {@code pathname} into a map
+   *
+   * @param pathname
```
Comment (rodsimpson) [inline library/src/main/java/YamlStringToSpecTreeConverter.java:30]: param description (i.e. - is the file local, http, etc)

File: library/src/main/java/YamlStringToSpecTreeConverter.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.util.LinkedHashMap;
+import org.yaml.snakeyaml.Yaml;
+
+/** Provides static functions for converting YAML files and strings into Maps */
+public class YamlStringToSpecTreeConverter {
+
+  /**
+   * deserializes a YAML file at the given {@code pathname} into a map
```
Comment (rodsimpson) [inline library/src/main/java/YamlStringToSpecTreeConverter.java:28]: Nit: capitalize sentence, add period.

File: library/src/main/java/YamlStringToSpecTreeConverter.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    File file = new File(pathname);
+    InputStream stream = new FileInputStream(file);
+
+    LinkedHashMap<String, Object> yamlMap = yaml.load(stream);
+
+    return yamlMap;
+  }
+
+  /**
+   * deserializes a YAML string {@code yamlString} into a map
```
Comment (rodsimpson) [inline library/src/main/java/YamlStringToSpecTreeConverter.java:47]: Nit: capitalize sentence, add period.

File: library/src/test/resources/order.yaml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+openapi:
+info:
+paths:
+  /pets:
+    get:
```
Comment (rodsimpson) [inline library/src/test/resources/order.yaml:5]: newline

**Comment by rkhaled98** [comment]
addressed comments above

File: library/src/main/java/YamlStringToSpecTreeConverter.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+import java.util.LinkedHashMap;
+import org.yaml.snakeyaml.Yaml;
```
Comment (ApigeeCorky) [inline library/src/main/java/YamlStringToSpecTreeConverter.java:22]: Not familiar with Snake Yaml, how did we decide on this?

Comment (rkhaled98) [reply library/src/main/java/YamlStringToSpecTreeConverter.java:22]: snakeyaml is actually provided by Jackson! It provides a nice way to deserialize a YAML file into the spec tree (LinkedHashMap)

**Comment by rkhaled98** [comment]
updated tests with feedback from https://github.com/googleinterns/spec-math/pull/14#discussion_r452541270

**Pull request merged**